### PR TITLE
Use typesafe network featuremaps instead of re-defining constants

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Network commissioning class was re-defining constants for feature maps.

#### Change overview
Switch to using bitflags of expected types, updated logic that relies on 'wifi driver' to use the driver to check not bitflags, use typesafe bitflag comparisons.

#### Testing
Commissioned a ESP32 devkit-c manually after the change (this is a device that reports wifi).

### CODE REVIEW

**Review by cecille — APPROVED** [review]

File: src/app/clusters/network-commissioning/network-commissioning.h [hunk 1]
```diff
@@ -66,16 +66,12 @@ class Instance : public CommandHandlerInterface,
...
-    static constexpr uint32_t kFeatureMapThread   = 0b0000'0010;
-    static constexpr uint32_t kFeatureMapEthernet = 0b0000'0100;
-
-    static constexpr uint32_t kFeatureMapWireless = 0b0000'0011;
-
     static void _OnCommissioningComplete(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg);
     void OnCommissioningComplete(CHIP_ERROR err);
 
-    const uint32_t mType;
+    using NetworkCommissioningFeature = chip::app::Clusters::NetworkCommissioning::NetworkCommissioningFeature;
```
Comment (bzbarsky-apple) [inline src/app/clusters/network-commissioning/network-commissioning.h:72]: Why is this needed?  This is all inside the `chip::app::Clusters::NetworkCommissioning` namespace, so you can just use `NetworkCommissioningFeature` without this `using` bit.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

File: src/app/clusters/network-commissioning/network-commissioning.h (hunk 1 above)
Comment (andy31415) [reply src/app/clusters/network-commissioning/network-commissioning.h]: Done.
